### PR TITLE
Fix UUID splitting

### PR DESCRIPTION
**Pull request by sparrc**

### CODE REVIEW

File: mpd/duration.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ var (
 var xmlDurationRegex = regexp.MustCompile(rStart + rDays + rTime + rHours + rMinutes + rSeconds + rEnd)
 
 func (d Duration) MarshalXMLAttr(name xml.Name) (xml.Attr, error) {
-	return xml.Attr{name, d.String()}, nil
+	return xml.Attr{Name: name, Value: d.String()}, nil
```
Comment (sparrc) [inline mpd/duration.go]: unrelated linting fix

File: mpd/mpd_test.go [hunk 2]
```diff
@@ -299,7 +299,7 @@ func TestAddNewContentProtectionRootErrorEmptyDefaultKID(t *testing.T) {
 	m := NewMPD(DASH_PROFILE_LIVE, VALID_MEDIA_PRESENTATION_DURATION, VALID_MIN_BUFFER_TIME)
 	s, _ := m.AddNewAdaptationSetVideo("7357", DASH_MIME_TYPE_VIDEO_MP4, VALID_SCAN_TYPE, VALID_SEGMENT_ALIGNMENT, VALID_START_WITH_SAP)
 
-	cp, err := s.AddNewContentProtectionRoot("")
+	cp, err := s.AddNewContentProtectionRoot2("")
```
Comment (stuarthicks) [inline mpd/mpd_test.go]: Can we come up with a more descriptive name please?

Comment (GeneticGenesis) [reply mpd/mpd_test.go]: AddNewContentProtectionRootThatDoesntMisformatUUIDs

Comment (sparrc) [reply mpd/mpd_test.go]: `AddNewContentProtectionRootFixedUUID`?

**Comment by mbowBC** [comment]
think a one test for the old AddNewContentProtectionRoot needs to be left in.
